app/post-contribution/route.js

```diff
@@ -17,7 +17,9 @@ export default Route.extend(AuthenticatedRouteMixin, {
   model({ date = moment().format('YYYY-MM-DD') }) {
     let user = this.get('currentUser.user')
 
-    return this.store.createRecord('contribution', { user, date })
+    let challenge = this.store.findRecord('challenge', 'current')
```

Comment:
Yes, the challenge relation is async, this means on a slow connection we *might* run the validations before the challenge is loaded (imagine f5 on the edit-contribution page).

Tests are indeed lacking ☹️